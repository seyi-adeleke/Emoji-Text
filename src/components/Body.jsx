import React from 'react';
import InputBox from './InputBox.jsx';
import TextBox from './TextBox.jsx';
import ButtonGroup from './ButtonGroup.jsx';
import PropTypes from 'prop-types';

class Body extends React.Component {
  constructor(props) {
    super(props);
    this.state = { text: '', emoji: 'ðŸ˜‚' };
    this.handleInput = this.handleInput.bind(this);
    this.handleEmoji = this.handleEmoji.bind(this);
  }

  handleInput(event) {
    this.setState({
      text: event
    });
  }

  handleEmoji(newEmoji) {
    this.setState({ emoji: newEmoji }, () => {
      // return (this.state.emoji);
    });
  }

  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col align-self-center">
              <h1 className="text-center">Hello world</h1>
              <InputBox getInput={this.handleInput} />
              <TextBox getText={this.state.text} emoji={this.state.emoji}/>
              <ButtonGroup getEmoji={this.handleEmoji}/>
          </div>
        </div>
      </div>
    );
  }
}

export default Body;
