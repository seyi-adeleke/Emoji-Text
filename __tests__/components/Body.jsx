import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Body from '../../src/components/Body.jsx';


describe('Body Component', () => {
  it('should match the Body snapshot', () => {
    const component = shallow(<Body />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
