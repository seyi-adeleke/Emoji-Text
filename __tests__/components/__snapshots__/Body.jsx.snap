// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Component should match the Body snapshot 1`] = `
<div
  className="columns"
>
  <div
    className="column is-half is-offset-one-quarter has-text-centered is-size-1-desktop is-size-1-mobile"
  >
    <h1
      className="has-text-weight-bold"
    >
      Emoji Text
    </h1>
    <InputBox
      getInput={[Function]}
    />
    <TextBox
      getText=""
    />
    <ButtonGoup
      getEmoji={[Function]}
    />
  </div>
</div>
`;
